From: =?utf-8?q?Lukas_M=C3=A4rdian?= <lukas.maerdian@canonical.com>
Date: Tue, 23 Mar 2021 10:59:02 +0100
Subject: tests:bonds: fix flaky resend_igmp test

---
 tests/integration/bonds.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/integration/bonds.py b/tests/integration/bonds.py
index fe84f7e..072f6b0 100644
--- a/tests/integration/bonds.py
+++ b/tests/integration/bonds.py
@@ -306,7 +306,7 @@ class _CommonTests():
             self.assertEqual(f.read().strip(), '15')
 
     def test_bond_resend_igmp(self):
-        self.setup_eth(None)
+        self.setup_eth(None, False)
         self.addCleanup(subprocess.call, ['ip', 'link', 'delete', 'mybond'], stderr=subprocess.DEVNULL)
         with open(self.config, 'w') as f:
             f.write('''network:
@@ -318,18 +318,18 @@ class _CommonTests():
       match: {name: %(e2c)s}
   bonds:
     mybond:
+      addresses: [192.168.9.9/24]
       interfaces: [ethbn, ethb2]
       parameters:
         mode: balance-rr
         mii-monitor-interval: 50s
         resend-igmp: 100
-      dhcp4: yes''' % {'r': self.backend, 'ec': self.dev_e_client, 'e2c': self.dev_e2_client})
+''' % {'r': self.backend, 'ec': self.dev_e_client, 'e2c': self.dev_e2_client})
         self.generate_and_settle()
         self.assert_iface_up(self.dev_e_client,
                              ['master mybond'],
                              ['inet '])
-        self.assert_iface_up('mybond',
-                             ['inet 192.168.5.[0-9]+/24'])
+        self.assert_iface_up('mybond', ['inet 192.168.9.9/24'])
         with open('/sys/class/net/mybond/bonding/slaves') as f:
             result = f.read().strip()
             self.assertIn(self.dev_e_client, result)
