From: =?utf-8?q?Lukas_M=C3=A4rdian?= <lukas.maerdian@canonical.com>
Date: Fri, 8 Jan 2021 09:28:23 +0100
Subject: tests:tunnels: improve test reliability

---
 tests/integration/tunnels.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/integration/tunnels.py b/tests/integration/tunnels.py
index 9299aa3..0bc6e58 100644
--- a/tests/integration/tunnels.py
+++ b/tests/integration/tunnels.py
@@ -123,7 +123,7 @@ class _CommonTests():
         self.assertIn("fwmark: 0x2a", out)
         self.assertIn("peer: M9nt4YujIOmNrRmpIRTmYSfMdrpvE7u6WkG8FY8WjG4=", out)
         self.assertIn("allowed ips: 20.20.20.0/24", out)
-        self.assertRegex(out, r'latest handshake: \d+ seconds? ago')
+        self.assertRegex(out, r'latest handshake: (\d+ seconds? ago|Now)')
         self.assertRegex(out, r'transfer: \d+ B received, \d+ B sent')
         self.assert_iface('wg0', ['inet 10.10.10.20/24'])
         # Verify client
