From: =?utf-8?q?Lukas_M=C3=A4rdian?= <lukas.maerdian@canonical.com>
Date: Fri, 9 Oct 2020 11:12:39 +0200
Subject: tests:ovs: fix OVS timeouts

---
 tests/integration/ovs.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/tests/integration/ovs.py b/tests/integration/ovs.py
index fb1cd36..9a75258 100644
--- a/tests/integration/ovs.py
+++ b/tests/integration/ovs.py
@@ -370,18 +370,15 @@ class _CommonTests():
         %(ec)s:
             addresses: [10.5.32.26/20]
             gateway4: 10.5.32.1
-            match:
-                macaddress: %(e_mac)s
             mtu: 1500
             nameservers:
                 addresses: [10.5.32.99]
                 search: [maas]
-            set-name: %(ec)s
     vlans:
         %(ec)s.21:
             id: 21
             link: %(ec)s
-            mtu: 1500''' % {'ec': self.dev_e_client, 'e_mac': self.dev_e_client_mac})
+            mtu: 1500''' % {'ec': self.dev_e_client})
         self.generate_and_settle()
         # Basic verification that the interfaces/ports are set up in OVS
         out = subprocess.check_output(['ovs-vsctl', 'show'], universal_newlines=True)
