Index: netplan.io-0.103/src/parse.h
===================================================================
--- netplan.io-0.103.orig/src/parse.h
+++ netplan.io-0.103/src/parse.h
@@ -402,6 +402,9 @@ struct net_definition {
     /* netplan-feature: activation-mode */
     char* activation_mode;  
 
+    /* configure without carrier */
+    gboolean ignore_carrier;
+
     /* offload options */
     gboolean receive_checksum_offload;
     gboolean transmit_checksum_offload;
